src/shogun/labels/LabelEncoder.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ *
+ */
+
+#ifndef _LABELENCODER__H__
+#define _LABELENCODER__H__
+
+#include <algorithm>
+#include <map>
+#include <memory>
+#include <set>
+#include <shogun/base/SGObject.h>
+#include <shogun/lib/SGVector.h>
+namespace shogun
+{
+
+	class LabelEncoder : public SGObject
+	{
+	public:
+		LabelEncoder() = default;
+
+		virtual ~LabelEncoder() = default;
+
+		/** Fit label encoder
+		 *
+		 * @param Target values.
+		 * @return SGVector which contains unique labels.
+		 */
+		virtual SGVector<float64_t>
+		fit(const std::shared_ptr<Labels>& labs) = 0;
+		/** Transform labels to normalized encoding.
+		 *
+		 * @param  Target values to be transformed.
+		 * @return Labels transformed to be normalized.
+		 */
+		virtual std::shared_ptr<Labels>
+		transform(const std::shared_ptr<Labels>& labs) = 0;
+		/** Transform labels back to original encoding.
+		 *
+		 * @param normailzed encoding labels
+		 * @return original encoding labels
+		 */
+		virtual std::shared_ptr<Labels>
+		inverse_transform(const std::shared_ptr<Labels>&) = 0;
+
+		/** Fit label encoder and return encoded labels.
+		 *
+		 * @param Target values.
+		 * @return Labels transformed to be normalized.
+		 */
+		virtual std::shared_ptr<Labels>
+		fit_transform(const std::shared_ptr<Labels>&) = 0;
+
+		virtual const char* get_name() const
+		{
+			return "LabelEncoder";
+		}
+
+		void set_print_warning(bool print_warning){
```

Comment:
Imo this shouldn’t be needed. The user should set this with log level, eg switch from warn to error

Reply:
++